pyposeidon/mgmsh.py

```diff
@@ -635,34 +634,53 @@ def gmsh_execute(**kwargs):
         else:
             dim = -2
 
+        gmsh_args = kwargs.get("gmsh_args", {})
+
+        gmsh_args.update({"tol": 1e-20})
+
+        gargs = ""
+        for k, v in list(gmsh_args.items()):
+            if k[0] != "-":
+                k = "-" + k
+            gargs = gargs + " ".join([k, str(v)]) + " "
+
         # execute gmsh
-        with subprocess.Popen(
-            #            ["{} -bin -tol {} {} {}".format(bin_path, 1e-20, dim, "mymesh.geo")],
-            ["{} -tol {} {} {}".format(bin_path, 1e-20, dim, "mymesh.geo")],
+        ex = subprocess.run(
```

Comment:
minor nitpick: At least to me, the reference to `ex` is not clear. "executable"? I would consider renaming to `proc` (from process) because that's what `run()` returns (a [CompletedProcess](https://docs.python.org/3/library/subprocess.html#subprocess.CompletedProcess) object to be more precise). or, if I wanted to be more specific, I would also use the executable name (e.g. `gmsh_proc`).